### PR TITLE
📚 Doc: Updates to Context documentation

### PR DESCRIPTION
**Pull request by gaby**
# Description

- Auto update the Context documentation using `o1-mini` from OpenAI

Related to #2712 

## Changes introduced
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [x] Examples: Provide examples demonstrating the new features or changes in action.

## Type of change
- [x] Documentation update (changes to documentation)

### CODE REVIEW

**Review by efectn — APPROVED** [review]

File: docs/api/ctx.md [hunk 1]
```diff
@@ -102,10 +101,10 @@ app.Get("/", func(c fiber.Ctx) error {
 
 ## App
 
-Returns the [\*App](ctx.md) reference so you could easily access all application settings.
+Returns the [*App](app.md) reference so you can easily access all application settings.
```
Comment (ReneWerner87) [inline docs/api/ctx.md:104]:
```suggestion
Returns the [\*App](app.md) reference so you can easily access all application settings.
```

File: docs/api/ctx.md [hunk 2]
```diff
@@ -1154,22 +1127,20 @@ app.Get("/", func(c fiber.Ctx) error {
...
 :::
 
 ## Params
 
-Method can be used to get the route parameters, you could pass an optional default value that will be returned if the param key does not exist.
+This method can be used to get the route parameters. You can pass an optional default value that will be returned if the param key does not exist.
 
 :::info
-Defaults to empty string \(`""`\), if the param **doesn't** exist.
+Defaults to an empty string (`""`) if the param **doesn't** exist.
```
Comment (ReneWerner87) [inline docs/api/ctx.md:1139]: not sure if we need the escaping for the parenthese
```suggestion
Defaults to an empty string \(`""`\) if the param **doesn't** exist.
```

File: docs/api/ctx.md [hunk 3]
```diff
@@ -1189,7 +1160,7 @@ app.Get("/user/*", func(c fiber.Ctx) error {
 })
 ```
 
-Unnamed route parameters\(\*, +\) can be fetched by the **character** and the **counter** in the route.
+Unnamed route parameters (*, +) can be fetched by the **character** and the **counter** in the route.
```
Comment (ReneWerner87) [inline docs/api/ctx.md:1163]:
```suggestion
Unnamed route parameters \(\*, +\) can be fetched by the **character** and the **counter** in the route.
```

File: docs/api/ctx.md [hunk 4]
```diff
@@ -1496,10 +1457,10 @@ app.Get("/", func(c fiber.Ctx) error {
 
 ## RequestCtx
 
-Returns [\*fasthttp.RequestCtx](https://godoc.org/github.com/valyala/fasthttp#RequestCtx) that is compatible with the context.Context interface that requires a deadline, a cancellation signal, and other values across API boundaries.
+Returns [*fasthttp.RequestCtx](https://pkg.go.dev/github.com/valyala/fasthttp#RequestCtx) that is compatible with the `context.Context` interface that requires a deadline, a cancellation signal, and other values across API boundaries.
```
Comment (ReneWerner87) [inline docs/api/ctx.md:1460]:
```suggestion
Returns [\*fasthttp.RequestCtx](https://pkg.go.dev/github.com/valyala/fasthttp#RequestCtx) that is compatible with the `context.Context` interface that requires a deadline, a cancellation signal, and other values across API boundaries.
```

File: docs/api/ctx.md [hunk 5]
```diff
@@ -1508,10 +1469,10 @@ Please read the [Fasthttp Documentation](https://pkg.go.dev/github.com/valyala/f
 
 ## Response
 
-Response return the [\*fasthttp.Response](https://godoc.org/github.com/valyala/fasthttp#Response) pointer
+Returns the [*fasthttp.Response](https://pkg.go.dev/github.com/valyala/fasthttp#Response) pointer.
```
Comment (ReneWerner87) [inline docs/api/ctx.md:1472]:
```suggestion
Returns the [\*fasthttp.Response](https://pkg.go.dev/github.com/valyala/fasthttp#Response) pointer.
```

File: docs/api/ctx.md [hunk 6]
```diff
@@ -2051,80 +2012,80 @@ app.Get("/", func(c fiber.Ctx) error {
...
-  fmt.Fprintf(c, "%s\n", "Hello, World!") // "Hello, World!Hello, World!"
+  return c.WriteString("Hello, World!")
+  // => "Hello, World!"
 })
 ```
 
 ## XHR
 
-A Boolean property, that is `true`, if the request’s [X-Requested-With](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) header field is [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), indicating that the request was issued by a client library \(such as [jQuery](https://api.jquery.com/jQuery.ajax/)\).
+A boolean property that is `true` if the request’s [X-Requested-With](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Requested-With) header field is `XMLHttpRequest`, indicating that the request was issued by a client library (such as [jQuery](https://api.jquery.com/jQuery.ajax/)).
```
Comment (ReneWerner87) [inline docs/api/ctx.md:2085]: image
```suggestion
A boolean property that is `true` if the request’s [X-Requested-With](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) header field is [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), indicating that the request was issued by a client library (such as [jQuery](https://api.jquery.com/jQuery.ajax/)).
```

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
some little AI mistakes

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**